### PR TITLE
Created Data Layer and UI Layer. Cleaned UI warnings.

### PR DESCRIPTION
**Pull request by ldanielcano**
- Moved fragments and activities to the UI layer.
- Moved Movie.kt and Movies.kt to the datalayer
- Cleaned pending warnings from the UI.

### CODE REVIEW

**Review by estebanhiguitad — CHANGES_REQUESTED** [review]
Agregar la capa de dominio. En la arquitectura es opcional, pero como objetivo de aprendizaje vamos a usarla.

**Comment by ldanielcano** [comment]
New changes done:

- Created a MovieModel to be used by the application
- Created a MovieData object
- Created MovieLocalDataSource interface and MovieLocalDataSourceImpl 
- Used MovieLocalDataSource for inversion of dependencies
- Created a Repository and transformed the MovieData list into a MovieModel list

**Review by ldanielcano — COMMENTED** [review]
Confirm receipt

File: IMDbApp/app/src/main/java/com/danielcano/imdbapp/uilayer/menu/SearchFragment.kt [hunk 1]
```diff
@@ -24,18 +25,22 @@ class SearchFragment : Fragment() {
...
         val movieListAdapter = MovieListAdapter(::showMovieDetails)
         movieList.adapter = movieListAdapter
 
-        val itemList = movies(this.resources)
+//        val itemList = requestMoviesSource(this.resources)
+//        movieListAdapter.submitList(itemList)
+        val itemList = MoviesRepositoryImpl(MoviesLocalDataSourceImpl(this.resources)).getMovies()
         movieListAdapter.submitList(itemList)
-
+//        val moviesObserver = Observer<List<MovieModel>>{
```
Comment (estebanhiguitad) [inline IMDbApp/app/src/main/java/com/danielcano/imdbapp/uilayer/menu/SearchFragment.kt]: eliminar código comentado

File: IMDbApp/app/src/main/java/com/danielcano/imdbapp/uilayer/menu/HomeFragment.kt [hunk 2]
```diff
@@ -23,17 +27,23 @@ class HomeFragment: Fragment() {
...
         movieList.layoutManager = LinearLayoutManager(requireContext(),LinearLayoutManager.HORIZONTAL,false)
-
         val movieListAdapter = MovieListAdapter(::showMovieDetails)
         movieList.adapter = movieListAdapter
 
-        val itemList = movies(this.resources)
+//        val itemList = movies(this.resources)
+        val itemList = MoviesRepositoryImpl(MoviesLocalDataSourceImpl(this.resources)).getMovies()
         movieListAdapter.submitList(itemList)
+//        val moviesObserver = Observer<List<MovieModel>>{
```
Comment (estebanhiguitad) [inline IMDbApp/app/src/main/java/com/danielcano/imdbapp/uilayer/menu/HomeFragment.kt]: eliminar código comentado en todos los lugares que aplique en esta clase

**Pull request closed without merging**